### DISCUSSION TITLE
golang project that injected skywalking-go-agent will froze when skywalking backend can not access

### DISCUSSION THREAD

**Discussion by GlqEason** [apache/skywalking#11416] [Q&A]
When the skywalking backend went down, the golang project that injected Skywalk-go-agent froze a moment later, and memory continued to rise. The project is a web server that uses the gin framework

**Comment by mrproliu** [comment]
When the go-agent fails to establish a connection with the Skywalking backend, it would cache monitoring data in memory. [This parameter can be used to change the size of its data volume.](https://github.com/apache/skywalking-go/blob/316433e71808ec6a130e81c287caa8b2a98f45a5/tools/go-agent/config/agent.default.yaml#L38C1-L39) You can try lowering this value to see if the memory usage will decrease.

**Reply by GlqEason** [reply]
I have tried changing the SW_AGENT_REPORTER_GRPC_MAX_SEND_QUEUE parameter and it does reduce the memory usage, but the problem of the main project getting stuck is still not resolved. The symptom is that my web server is unable to provide service and no logs are being output.  I suspect that the gin plugin's modification of gin is the cause.

**Reply by mrproliu** [reply]
Have you tried to dump all the goroutine stack when stuck? More information is needed for this kind of question.

**Reply by GlqEason** [reply]
not yet and how can i dump goroutine stack ? PrintStack in gin middleware ?  pprof also can not return data

**Reply by mrproliu** [reply]
You can startup a pprof HTTP server, and dump all goroutine stack through HTTP request.
